FirebaseAuth/Sources/AuthProvider/OAuth/FIROAuthProvider.m

```diff
@@ -358,16 +367,54 @@ - (void)getHeadFulLiteURLWithEventID:(NSString *)eventID
                                            [NSString stringWithFormat:kHeadfulLiteURLStringFormat,
                                                                       authDomain, argumentsString];
                                      }
-                                     if (completion) {
-                                       NSCharacterSet *set =
-                                           [NSCharacterSet URLFragmentAllowedCharacterSet];
-                                       completion(
-                                           [NSURL
-                                               URLWithString:
-                                                   [URLString
-                                                       stringByAddingPercentEncodingWithAllowedCharacters:
-                                                           set]],
-                                           nil);
+                                     if (appCheck) {
+                                       [appCheck
+                                           getTokenForcingRefresh:false
+                                                       completion:^(
+                                                           id<FIRAppCheckTokenResultInterop> _Nonnull tokenResult) {
+                                                         if (tokenResult.error) {
+                                                           FIRLogWarning(
+                                                               kFIRLoggerAuth, @"I-AUT000018",
+                                                               @"Error getting App Check token; "
+                                                               @"using placeholder token "
+                                                               @"instead. Error: %@",
+                                                               tokenResult.error);
+                                                         }
+                                                         NSString *appCheckTokenFragments =
+                                                             [kAppCheckURLFragmentKey
+                                                                 stringByAppendingString:
+                                                                     tokenResult.token];
+                                                         NSString *URLStringWithAppCheckToken =
+                                                             [URLString stringByAppendingString:
+                                                                            appCheckTokenFragments];
+                                                         NSLog(@"URLString with "
+                                                               @"appCheckTokenFragment: %@",
+                                                               URLString);
+                                                         if (completion) {
```

Comment:
@renkelvin Do you know if there's any better way to handle the async flow here. Because getTokenForcingRefresh is also an async function.

Reply:
Hm, I don't see a better way in Objc C other than nested callbacks.